### PR TITLE
Add local tarball install #5812

### PR DESCRIPTION
**Pull request by axlEscalada**
…all it, test for local tarball

### What does this PR do?
- Check if dependency is a tarball before check if it is an npm package
- Alloc `url` to avoid losing tiny string when task is read.

Related to: https://github.com/oven-sh/bun/issues/5812

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

- [ ] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [x] I or my editor ran `zig fmt` on the changed files
- [ ] I included a test for the new code, or an existing test covers it
- [ ] JSValue used outside outside of the stack is either wrapped in a JSC.Strong or is JSValueProtect'ed

### CODE REVIEW

File: test/cli/install/bun-add.test.ts [hunk 1]
```diff
@@ -1627,3 +1627,53 @@ it("should add dependency alongside peerDependencies", async () => {
...
+    JSON.stringify({
+      name: "foo",
+      version: "0.0.1",
+    }),
+  );
+  const tarball = "baz-0.0.3.tgz"
+  await copyFile(tarball, join(package_dir, tarball));
+  const { stdout, stderr, exited } = spawn({
+    // cmd: ["pwd"],
+    cmd: ["/Users/axel.escalada/bun-dev/packages/debug-bun-darwin-aarch64/bun-debug", "add", tarball],
```
Comment (Jarred-Sumner) [inline test/cli/install/bun-add.test.ts]:
```suggestion
    cmd: [bunExe(), "add", tarball],
```

**Review by Jarred-Sumner — APPROVED** [review]
Thank you

**Comment by Jarred-Sumner** [comment]
looks like the file path to the tarball is not quite right
```
  ENOENT: No such file or directory
     path: "baz-0.0.3.tgz"
   syscall: "stat"
     errno: -2
  
  ✗ should add local tarball dependency [0.56ms]
  ```

**Comment by axlEscalada** [comment]
> looks like the file path to the tarball is not quite right
> 
> ```
>   ENOENT: No such file or directory
>      path: "baz-0.0.3.tgz"
>    syscall: "stat"
>      errno: -2
>   
>   ✗ should add local tarball dependency [0.56ms]
> ```

Ok, I was running the test inside of the test folder and it was trying to copy the file using a relative path. Now the test uses an absolute path to copy the tarball file, so the test passes from anywhere.

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**